packages/api/core/src/util/forge-config.ts

```diff
@@ -6,13 +6,24 @@ import * as interpret from 'interpret';
 import { template } from 'lodash';
 import * as rechoir from 'rechoir';
 
-// eslint-disable-next-line n/no-missing-import
-import { dynamicImportMaybe } from '../../helper/dynamic-import.js';
+import { dynamicImportMaybe } from '../../helper/dynamic-import';
 
 import { runMutatingHook } from './hook';
 import PluginInterface from './plugin-interface';
 import { readRawPackageJson } from './read-package-json';
 
+/* eslint-disable @typescript-eslint/no-require-imports */
+// TSX imports only work with Node16 resolution while we're still on CommonJS resolution in tsconfig.base.json
+// However, all of Vite's entire TS types break when using CommonJS with Node16 resolution, it's more approachable
+// to use `require` directly for the time being.
+const tsxCJS = require('tsx/cjs/api');
+const tsxESM = require('tsx/esm/api');
+/* eslint-enable @typescript-eslint/no-require-imports */
+
+// Register tsx enhancements
+const unregisterCJS = tsxCJS.register();
+const unregisterESM = tsxESM.register();
```

Comment:
We should do this _just_ before importing, not on module load

Reply:
[6b65915](https://github.com/electron/forge/pull/3881/commits/6b65915d84dd8c137879b01741734ef314458558)